octobot_trading/exchanges/connectors/ccxt_exchange.py

```
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
-                # get a cancelled order).
-                return True
+            cancel_resp_message = f" - Cancel response: {cancel_resp or 'no response'}"
+            is_canceled, _ = await self.check_if_canceled(order_id, symbol)
+            if is_canceled:
+                success = True
+                return success, None
+            await asyncio.sleep(20)  # try again - some exchanges need a while to update the order (bybit for example)
+            is_canceled, _ = await self.check_if_canceled(order_id, symbol, cancel_resp_message)
+            return is_canceled, None
         except ccxt.OrderNotFound:
-            self.logger.error(f"Trying to cancel order with id {order_id} but order was not found")
+            return success, f"Trying to cancel order (id {order_id}) with cancel_order_default " \
+                            f"but order was not found{cancel_resp_message or ''}\n"
         except (ccxt.NotSupported, octobot_trading.errors.NotSupported) as e:
-            raise octobot_trading.errors.NotSupported from e
+            return success, f"cancel_order_default is not supported. Error: {e}{cancel_resp_message or ''}\n"
         except Exception as e:
-            self.logger.exception(e, True, f"Order {order_id} failed to cancel | {e} ({e.__class__.__name__})")
-        return cancel_resp is not None
+            return success, f"Order {order_id} failed to cancel using " \
+                            f"| {e} ({e.__class__.__name__}){cancel_resp_message or ''}\n"
+
+    async def check_if_canceled(self, order_id, symbol, cancel_resp_message="") -> typing.Tuple[bool, str or None]:
+        success = False
+        try:
+            # check if canceled
+            cancelled_order = await self.get_order(order_id, symbol=symbol)
+            if cancelled_order:
+                if personal_data.parse_is_cancelled(cancelled_order):
+                    success = True
+                    return success, None
```

I think we should just return a bool. We should raise an exception when the order is still uncanceled that can be catched by the calling method

---

i changed it, now its way better